src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let ACTION_REQUIRED = '';
+  const stateCopy = { ...state };
+  const ARRAY = [Object.assign({}, stateCopy)];
+
+  for (const action of actions) {
+    ACTION_REQUIRED = action.type;
+    let elementArray = Object.assign({}, ARRAY[ARRAY.length - 1]);
+
+    switch (ACTION_REQUIRED) {
+      case 'addProperties':
+        elementArray = Object.assign(elementArray, action.extraData)
```

Comment:
The elementArray = Object.assign(elementArray, action.extraData) statement can be simplified to Object.assign(elementArray, action.extraData). No need to reassign the value to elementArray

Reply:
Fixed.